src/patch_parse.c

```diff
@@ -109,7 +109,10 @@ static int header_path_len(git_patch_parse_ctx *ctx)
 	size_t len;
 
 	for (len = quoted; len < ctx->line_len; len++) {
-		if (!quoted && git__isspace(ctx->line[len]))
+		if (!quoted &&
+		    (!git__strncmp(ctx->line + len, " b/", strlen(" b/")) ||
+		     !git__strncmp(ctx->line + len, " \"b/", strlen(" \"b/")) ||
```

Comment:
Except that a patch needn't _have_ `+++` and `---` lines.

```
diff --git foo bar foo bar
old mode 100755
new mode 100644
```

(Note: git cannot apply this patch.)